### ISSUE TITLE
Associations nan_replace_value affects original input data

### ISSUE THREAD

**Issue by daanknoors** [shakedzy/dython#82]
Thanks for all the work on this package - it's been really helpful.

I noticed a minor bug. When running associations or compute_associations the nan_replace_strategy affects the original input data.  Hence, if you have a dataframe with NaN values in certain columns, these get replaced in the original input. It would be better to work on a copy of the original data, instead of processing the data inplace.

Pandas gives the following warning:
 
```
SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame
```

**Comment by shakedzy** [comment]
@daanknoors thanks for this - you're right. I think there's an easy fix for it.

**Comment by shakedzy** [comment]
Fixed in version 0.6.6

**Comment by daanknoors** [comment]
Thanks for the quick fix! Tested it and seems to work correctly.
